packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -226,18 +221,20 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
   void _flushPointerEventQueue() {
     assert(!locked);
 
+    while (_pendingPointerEvents.isNotEmpty)
+      handlePointerEvent(_pendingPointerEvents.removeFirst());
+
     if (resamplingEnabled) {
-      _resampler.addOrDispatchAll(_pendingPointerEvents);
-      _resampler.sample(samplingOffset);
-      return;
+      triggerResample();
     }
+  }
 
-    // Stop resampler if resampling is not enabled. This is a no-op if
-    // resampling was never enabled.
-    _resampler.stop();
-
-    while (_pendingPointerEvents.isNotEmpty)
-      handlePointerEvent(_pendingPointerEvents.removeFirst());
+  /// to force resampler for triggering a resample operation.
+  ///
+  /// This should only be called when resampling is enabled.
+  void triggerResample() {
```

Comment:
I'm still feeling this is a workaround, see also https://github.com/flutter/flutter/pull/60558#discussion_r496235528 
This is a public method because it's needed in `WidgetTester`.

Reply:
why is this needed in WidgetTester? If needed then maybe this is find but ideally the name would just be "sample()" instead of triggerResample.

Reply:
> why is this needed in WidgetTester? If needed then maybe this is find but ideally the name would just be "sample()" instead of triggerResample.

Because `triggerResample` is called in `_flushPointerEventQueue`, which is before input injection in `WidgetTester`. `triggerResample` is in `_flushPointerEventQueue` because otherwise when `_pendingPointerEvents` is empty, we still need to call `_resampler.sample` for updated resampled event (see https://github.com/flutter/flutter/pull/60558#discussion_r496235528) @dreveman

Reply:
As for the name, I'm feeling `binding.sample` is confusing because binding is not the subject of `sample`...

Reply:
Thank to @liyuqian 's [comment](https://github.com/flutter/flutter/pull/60558/files/b7802f69f91e1b5466d6b5b58e98193160964ca5#r499014047) I realize this `triggerResample` is kind of duplicate with `_handleSampleTimeChanged`. `triggerResample` is removed in the new commit.